airflow-core/src/airflow/ui/src/pages/Dashboard/PoolSummary/PoolSummary.tsx

```diff
@@ -0,0 +1,113 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { Box, Heading, Flex, Skeleton } from "@chakra-ui/react";
+import { BiTargetLock } from "react-icons/bi";
+
+import { usePoolServiceGetPools } from "openapi/queries/queries";
+import { Tooltip } from "src/components/ui";
+import { capitalize } from "src/utils";
+import { slots } from "src/utils/slots";
+
+export const PoolSummary = () => {
+  const { data, isLoading } = usePoolServiceGetPools();
+  const pools = data?.pools ?? [];
+  const totalSlots = pools.reduce((sum, pool) => sum + pool.slots, 0);
+
+  const slotTotals = {
+    deferred_slots: 0,
+    occupied_slots: 0,
+    open_slots: 0,
+    queued_slots: 0,
+    running_slots: 0,
+    scheduled_slots: 0,
+  };
+
+  const poolsWithSlotType = {
+    deferred_slots: 0,
+    occupied_slots: 0,
+    open_slots: 0,
+    queued_slots: 0,
+    running_slots: 0,
+    scheduled_slots: 0,
+  };
+
+  pools.forEach((pool) => {
+    Object.keys(slotTotals).forEach((slotKey) => {
+      const typedKey = slotKey as keyof typeof slotTotals;
+      const slotValue = pool[typedKey];
+
+      if (typeof slotValue === "number" && slotValue > 0) {
```

Comment:
Why this check ? Return type is `number` for all of those fields.

Reply:
Yes some local type should also avoid the type casting. (maybe something with Pick / Omit indeed)